Dockerfile

```diff
@@ -1,66 +1,96 @@
   && apt-get update -qqy \
   && apt-get -qqy --no-install-recommends install firefox \
+  && apt-get install libdbus-glib-1-2 \
   && rm -rf /var/lib/apt/lists/* /var/cache/apt/* \
-  && wget --no-verbose -O /tmp/firefox.tar.bz2 $FIREFOX_DOWNLOAD_URL \
+  && wget --no-verbose -O /tmp/firefox.tar.bz2 "$FIREFOX_DOWNLOAD_URL" \
   && apt-get -y purge firefox \
   && rm -rf /opt/firefox \
   && tar -C /opt -xjf /tmp/firefox.tar.bz2 \
   && rm /tmp/firefox.tar.bz2 \
   && mv /opt/firefox /opt/firefox-$FIREFOX_VERSION \
-  && ln -fs /opt/firefox-$FIREFOX_VERSION/firefox /usr/bin/firefox
+  && ln -fs /opt/firefox-"$FIREFOX_VERSION"/firefox /usr/bin/firefox \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
 
 #Geckodriver
-ARG GECKODRIVER_VERSION=latest
-RUN GK_VERSION=$(if [ ${GECKODRIVER_VERSION:-latest} = "latest" ]; then echo $(wget -qO- "https://api.github.com/repos/mozilla/geckodriver/releases/latest" | grep '"tag_name":' | sed -E 's/.*"v([0-9.]+)".*/\1/'); else echo $GECKODRIVER_VERSION; fi) \
-  && echo "Using GeckoDriver version: "$GK_VERSION \
+ARG GECKODRIVER_VERSION=0.32.2
+RUN GK_VERSION="$(if [ "${GECKODRIVER_VERSION:-latest}" = "latest" ]; then echo "$(wget -qO- 'https://api.github.com/repos/mozilla/geckodriver/releases/latest' | grep '\"tag_name\":' | sed -E 's/.*\"v([0-9.]+)\".*/\1/')"; else echo "$GECKODRIVER_VERSION"; fi)" \
+  && echo "Using GeckoDriver version: ""$GK_VERSION" \
   && wget --no-verbose -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GK_VERSION/geckodriver-v$GK_VERSION-linux64.tar.gz \
   && rm -rf /opt/geckodriver \
   && tar -C /opt -zxf /tmp/geckodriver.tar.gz \
   && rm /tmp/geckodriver.tar.gz \
-  && mv /opt/geckodriver /opt/geckodriver-$GK_VERSION \
-  && chmod 755 /opt/geckodriver-$GK_VERSION \
-  && ln -fs /opt/geckodriver-$GK_VERSION /usr/bin/geckodriver
+  && mv /opt/geckodriver /opt/geckodriver-"$GK_VERSION" \
+  && chmod 755 /opt/geckodriver-"$GK_VERSION" \
+  && ln -fs /opt/geckodriver-"$GK_VERSION" /usr/bin/geckodriver
 
 # Python 3.5 and Python Pip
 RUN apt-get update
-RUN apt-get install -y \
+RUN apt-get install -y --no-install-recommends \
     python3 \
     python3-setuptools \
-    python3-pip
\ No newline at end of file
+    python3-pip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Creating a new directory
+RUN mkdir /shell_script
+
+# Copying shell script to directory
+COPY entrypoint.sh /shell_script
+
+# Setting the working directory
+WORKDIR /shell_script
+
+# Setting the entry point
+ENTRYPOINT ["/bin/bash", "/shell_script/entrypoint.sh"]
+
+# Setting the default command to be run in the container
+CMD ["sh", "-c", "Xvfb :20 -screen 0 1366x768x16 & x11vnc -passwd password -display :20 -N -forever"]
```

Comment:
@Archana-GS - what is this 'password'? Is that the default that is set by which ever vnc server you are using?

Reply:
Yes it will set as default when ever vnc server is up